Sources/NIO/SocketChannel.swift

```diff
@@ -489,15 +500,28 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
         var rawAddressLength = socklen_t(MemoryLayout<sockaddr_storage>.size)
         var buffer = self.recvAllocator.buffer(allocator: self.allocator)
         var readResult = ReadResult.none
+        
+        let controlBytes = self.reportExplicitCongestionNotifications ?
+                // Guess at max 4 int32 payload messages.
+                UnsafeMutableRawBufferPointer.allocate(
```

Comment:
This allocation can actually be hoisted up to be on the event loop. We do this already for writing (see `SelectableEventLoop` which has a number of these pointers). As only one read can be active on an event loop at any given time we can guarantee that these pointers will not be clobbered during the read operation.